### PR TITLE
disable azure rs function through azure configs

### PR DESCRIPTION
**Pull request by zdeveloper**
## Related Issue or Background Info

- Looks like the way to disable the azure cloud function is via app config

this is auto generated from the Azure portal
image

reference:
https://docs.microsoft.com/en-us/azure/azure-functions/disable-function?tabs=portal

### Infrastructure
- [x] **Consult the results of the `terraform-plan` job inside the "Terraform Checks" workflow run for this PR. Confirm that there are no unexpected changes!**

### Design
- [x] Any UI/UX changes have a designer as a reviewer, and changes have been approved
- [x] Any large-scale changes have been deployed to `test`, `dev`, or `pentest` and smoke-tested by both the engineering and design teams

### Content
- [x] Any content changes (including new error messages) have been approved by content team

### Support
- [x] Any changes that might generate new support requests have been flagged to the support team
- [x] Any changes to support infrastructure have been demo'd to support team

### Testing
- [x] Includes a summary of what a code reviewer should verify

### Changes are Backwards Compatible
- [x] Database changes are submitted as a separate PR
  - [ ] Any new tables that do not contain PII are accompanied by a GRANT SELECT to the no-PHI user
  - [ ] Any changes to tables that have custom no-PHI views are accompanied by changes to those views
        (including re-granting permission to the no-PHI user if need be)
  - [ ] Liquibase rollback has been tested locally using `./gradlew liquibaseRollbackSQL` or `liquibaseRollback`
  - [ ] Each new changeset has a corresponding [tag](https://docs.liquibase.com/change-types/community/tag-database.html)
- [x] GraphQL schema changes are backward compatible with older version of the front-end

### Security
- [x] Changes with security implications have been approved by a security engineer (changes to  authentication, encryption, handling of PII, etc.)
- [x] Any dependencies introduced have been vetted and discussed

## Cloud
- [x] DevOps team has been notified if PR requires ops support
- [x] If there are changes that cannot be tested locally, this has been deployed to our Azure `test`, `dev`, or `pentest` environment for verification

### CODE REVIEW

**Review by nathancrtr — APPROVED** [review]
🚢

File: ops/services/app_functions/report_stream_batched_publisher/infra/main.tf [hunk 1]
```diff
@@ -68,25 +68,26 @@ resource "azurerm_function_app" "functions" {
...
+    AZ_STORAGE_QUEUE_CXN_STRING                          = data.azurerm_storage_account.app.primary_connection_string
+    TEST_EVENT_QUEUE_NAME                                = var.test_event_queue_name
+    REPORTING_EXCEPTION_QUEUE_NAME                       = var.reporting_exception_queue_name
+    REPORT_STREAM_URL                                    = local.report_stream_url
+    REPORT_STREAM_TOKEN                                  = "@Microsoft.KeyVault(SecretUri=${data.azurerm_key_vault_secret.datahub_api_key.id})"
+    REPORT_STREAM_BATCH_MINIMUM                          = var.report_stream_batch_minimum
+    REPORT_STREAM_BATCH_MAXIMUM                          = var.report_stream_batch_maximum
+    SIMPLE_REPORT_CB_URL                                 = local.simple_report_callback_url
+    SIMPLE_REPORT_CB_TOKEN                               = "@Microsoft.KeyVault(SecretUri=${data.azurerm_key_vault_secret.simple_report_callback_token.id})"
+    "AzureWebJobs.ReportStreamExceptionHandler.Disabled" = "1"
```
Comment (zdeveloper) [inline ops/services/app_functions/report_stream_batched_publisher/infra/main.tf:90]: Only added this line, the others are a format change

Comment (jdorothy) [reply ops/services/app_functions/report_stream_batched_publisher/infra/main.tf:90]: I think technically you don't have to quote this but I think it works either way. Did you get a chance to deploy this to a test env? I'm curious how the env var shows up like this.

Comment (zdeveloper) [reply ops/services/app_functions/report_stream_batched_publisher/infra/main.tf:90]: I actually do, I tried it without quotes and TF complained about not being able to fine `AzureWebJobs` resource

Comment (zdeveloper) [reply ops/services/app_functions/report_stream_batched_publisher/infra/main.tf:90]: Before on test
Screenshot from 2022-02-17 15-48-00

After on test
image

**Review by emmastephenson — APPROVED** [review]

**Pull request merged**